### PR TITLE
consumer support update stats with specified stats

### LINKED ISSUE #8949: consumer support update stats with specified stats
**Is your enhancement request related to a problem? Please describe.**
In the KoP scenario, raw data is consumed by reading bookie directly, which could cause no metrics data for consumers.

**Describe the solution you'd like**
we can expose `updateStats` with `ConsumerStats` param, which could update stats specified stats.

**Additional context**
the related issue in KoP is https://github.com/streamnative/kop/pull/263

### PR DESCRIPTION
**Pull request by dockerzhang**
Fixes #8949 

### Motivation
reuse pulsar metric for KoP

### Modifications

expose updateStats with ConsumerStats param, which could update stats specified stats.

### Verifying this change

- [√] Make sure that the change passes the CI checks.

### CODE REVIEW

**Comment by sijie** [comment]
@BewareMyPower Can you review this pull request?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java [hunk 1]
```diff
@@ -493,4 +494,29 @@ public void testOnlyCloseActiveConsumerForSingleActiveConsumerDispatcherWhenSeek
         }
         assertTrue(hasConsumerNotDisconnected);
     }
+
+    @Test
+    public void testUpdateStatsForSingleActiveConsumerDispatcherWhenSeek() throws Exception {
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java:499]: Is there anything related to seek? IMO, the test should move to org.apache.pulsar.broker.stats.ConsumerStatsTest

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java (hunk 1 above)
Comment (dockerzhang) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java:499]: @BewareMyPower I thought it's related to subscription consumer. You are right, it's the best to move to `org.apache.pulsar.broker.stats.ConsumerStatsTest`

**Review by BewareMyPower — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**